OpenTelemetry.proj

```diff
@@ -10,6 +10,11 @@
     <SolutionProjects Remove="src\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj" />
     <PackProjects Remove="src\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc\OpenTelemetry.Exporter.OpenTelemetryProtocol.Grpc.csproj" />
 
+    <!-- Skip building these projects when publish packages workflow runs as these projects need to dependupon instrumentation libraries taking a project reference
+    instead of  package reference on API/SDK-->
+    <SolutionProjects Remove="examples\**\*.csproj" />
+    <SolutionProjects Remove="test\Benchmarks\Benchmarks.csproj" />
```

Comment:
Does this work?

```suggestion
    <SolutionProjects Remove="examples\**\*.csproj" Condition="'$(RunningDotNetPack)' == 'true'" />
    <SolutionProjects Remove="test\Benchmarks\Benchmarks.csproj" Condition="'$(RunningDotNetPack)' == 'true'" />
```

Reply:
That might work but `OpenTelemetry.proj` is only used in the publish-packages workflow anyway.

Reply:
I approved to unblock you from doing the release, but I think adding this condition might be good since we might end up using this proj for other purposes. Another idea might be to change the name of the file to something like OpenTelemetryPack.proj if we think this will only ever be used for pack/publish.